js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useStaticSetFilter.tsx

```diff
@@ -69,13 +69,20 @@ export function useStaticSetFilter<TValue>({
   // This filter can be used as both a controlled and an uncontrolled component necessitating an innerState for the uncontrolled case.
   const [innerState, setState] = useState(() => new Set(state || []));
 
-  useEffect(() => {
-    onStateChanged?.(innerState);
+  const isFirstRenderRef = useRef(true);
+
+  useLayoutEffect(() => {
+    if (!isFirstRenderRef.current) {
+      onStateChanged?.(innerState);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [innerState]);
 
   useEffect(() => {
-    setState(state ? new Set(state) : new Set());
+    if (!isFirstRenderRef.current) {
+      setState(state ? new Set(state) : new Set());
+    }
+    isFirstRenderRef.current = false;
```

Comment:
This seems somewhat backwards -- "if not first render, setState(state)" seems like it will cause an infinite loop, since setState(state) triggers the deps. This would be more harmless if the if statement on 82 was flipped?

Reply:
`state` is the state passed in as a prop.
`innerState` is the state the component keeps track of in component state

`setState` is actually `setInnerState`. I'll update the naming here because I agree it's confusing.

Reply:
On the first render we don't need the extra `setState` since we default to `innerState` to whatever was passed in for`state`. So this was just getting rid of an unnecessary re-render that was causing an infinite loop when combined with passing a new empty array reference on every render.